Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -254,4 +254,89 @@ function Add-CustomCertificates($serviceName, $customCertificatesMap)
             Write-Verbose "Deleted the temp file $tmpFile.FullName"
         }
     }
+}
+function Validate-ServiceStatus ($ServiceName, $Slot)
+{
+		$RoleInstances=$null;
+		$deployment =$null;
+	    $retryCount = 0;
+		$readyRoleInstanceCount =0;
+	    $timeout = 0;
+	try{
+		Write-Verbose "First check for  existence of Azure Cloud Service with Service Name: $ServiceName "
+		if (Test-AzureName -Service -Name $ServiceName)
+		{
+			do
+			{
+				Write-Host "##[command]Get-AzureDeployment -ServiceName $ServiceName -Slot $Slot -ErrorAction SilentlyContinue -ErrorVariable azureDeploymentError"
+				$azureDeployment = Get-AzureDeployment -ServiceName $ServiceName -Slot $Slot -ErrorAction SilentlyContinue -ErrorVariable azureDeploymentError
```

Comment:
If Get-AzureDeployment cmdlet throws an exception, will $azureDeployment will still have not null value.